Uyanda.Coffee.Persistence/Accessors/BeverageAccessor.cs

```diff
@@ -67,56 +67,56 @@ public async Task<InvoiceModel> PurchaseAsync(IEnumerable<LineItemModel> lineIte
+        public async Task UpsertBeverageSizeCostAsync(BeverageSizeCostModel price)
         {
-          
-            var beverageDictionary = await localDbContext.Beverages.AsNoTracking()
-                .Where(d => d.Name == price.Beverage.Name)
-                .Select(c => new { c.Id, c.Name }).ToDictionaryAsync(c=>c.Id, c=>c.Name);
+          //update price
 
-            if (!beverageDictionary.ContainsValue(price.Beverage.Name))
+            
+                
+            if (price.Id > 0)
             {
-                localDbContext.BeverageCost.Add(ToEntity(price));
+                var beveragePrice = await localDbContext.BeverageCost
+                    .Where(c => c.Id == price.Id)
+                    .Include(d => d.Beverage)
+                    .SingleAsync();
 
-                await localDbContext.SaveChangesAsync();
+                beveragePrice.Beverage.Name = price.Beverage.Name;
+                beveragePrice.Cost = price.Cost;
+                beveragePrice.Beverage.IsActive = price.Beverage.IsActive;
+                beveragePrice.BeverageSizeId = price.BeverageSizeId;
 
-                var result = await localDbContext.BeverageCost.AsNoTracking().Select(c => c).ToArrayAsync();
+                await localDbContext.SaveChangesAsync();
 
-                return result.Select(ToModel).ToArray();
+                return;
             }
 
-            var beverageSizeDictionary = localDbContext.BeverageCost.AsNoTracking()
-                .Where(x => x.Beverage.Name == price.Beverage.Name)
-                .Select(c => new { c.BeverageSizeId, c.BeverageSize.Name }).ToDictionary(b => b.BeverageSizeId, b => b.Name);
 
-            if(beverageSizeDictionary.ContainsKey(price.BeverageSizeId))
+            if(price.BeverageId == 0)
             {
+                await localDbContext.BeverageCost.AddAsync(ToEntity(price));
 
-                var record = await localDbContext.BeverageCost
-                    .SingleOrDefaultAsync(c => c.BeverageSizeId == price.BeverageSizeId && c.Beverage.Name == price.Beverage.Name);
+                localDbContext.SaveChanges();
 
-                record.Cost = price.Cost;
-
-                await localDbContext.SaveChangesAsync();
-                    
+                return;
             }
-            else
-            {
 
-                price.BeverageId = beverageDictionary.FirstOrDefault(x => x.Value == price.Beverage.Name).Key;
+            var isBeverageAvailable = await localDbContext.BeverageCost.AsNoTracking()
+                .Where(c => c.BeverageId == price.BeverageId)
+                .AnyAsync();
 
+            if(isBeverageAvailable)
+            {
                 price.Beverage = null;
-                    
-                await localDbContext.AddAsync(ToEntity(price));
+
+                await localDbContext.BeverageCost.AddAsync(ToEntity(price));
 
                 await localDbContext.SaveChangesAsync();
-               
-            }
 
+                return;
```

Comment:
While it is not necessary now. . . It is worth never to swallow errors. So returning an error message/logging the fact that a none-existing Id was given could save you a lot of trouble.